src/Entity/Character.cpp

```diff
@@ -54,34 +55,51 @@ void Character::CharacterUpdate(flecs::entity characterEntity, TransformComponen
 	
 	//Changing orientation of the character depending on where it's moving
-	direction.x < 0.f ? transformComponent.mScale.x = -1.f : transformComponent.mScale.x = 1.f;
+	direction.x < 0.f ? transformComponent.mScale.x = -fabs(transformComponent.mScale.x) : transformComponent.mScale.x = fabs(transformComponent.mScale.x);
 	
 	//Check for map bounds
-	flecs::entity mapEntity = characterEntity.world().component<MapTag>().target<MapTag>();
-	Texture2D mapTexture = mapEntity.get_ref<TextureComponent>()->mTexture;
+	const flecs::entity mapEntity = characterEntity.world().component<MapTag>().target<MapTag>();
+	const Texture2D mapTexture = mapEntity.get_ref<TextureComponent>()->mTexture;
 
-	// Calculate new position
-	Vector2 newPosition = Vector2Add(transformComponent.mPosition, Vector2Scale(Vector2Normalize(direction), CHARACTER_SPEED * GetFrameTime()));
+	const Vector2 newPosition = Vector2Add(transformComponent.mPosition, Vector2Scale(Vector2Normalize(direction), CHARACTER_SPEED * GetFrameTime()));
 
 	// Check if the new position is within map boundaries 
-	//QUESTION: if i want this as function, would it be a system on map? how do i then call one?
+	//TODO: Write a function to do it in map namespace? if i find myself using same code for towers/enemies
 	if (newPosition.x >= 0 && newPosition.x <= mapTexture.width-CHARACTER_WIDTH &&
 		newPosition.y >= 0 && newPosition.y <= mapTexture.height-CHARACTER_HEIGHT)
 	{
-		  // Create a new rectangle that represents where the character would move
-		Rectangle newCharacterRect = {newPosition.x, newPosition.y, collisionComponent.mCollisionRect.width, collisionComponent.mCollisionRect.height};
+		const Rectangle newCharacterRect {newPosition.x, newPosition.y, collisionComponent.mRectScale.x, collisionComponent.mRectScale.y};
+
+		bool hasCollided{false};
+		//auto q = ecs.query<CollisionComponent, TransformComponent>();
+		//q.each([&hasCollided,&newCharacterRect, &characterEntity](flecs::entity e, CollisionComponent& otherCollisionComponent, TransformComponent& otherTransformComponent){
+		//	// Avoid computing a collision detection if we already detected a collision.
+		//	if (hasCollided)
+		//		return;
+		//	
+		//	const Rectangle otherRect {otherTransformComponent.mPosition.x, otherTransformComponent.mPosition.y, otherCollisionComponent.mRectScale.x, otherCollisionComponent.mRectScale.y};
+		//	if (e != characterEntity && CheckCollisionRecs(newCharacterRect, otherRect))
+		//	{
+		//		hasCollided = true;
+		//	}
+		//});
 
-		bool isCollision = false;
+		//QUESTION: I couldnt pass two components into .each query. I tried to create query and use it, it leads to a crash. Apparently I cant create a query inside a system? 
+		//passing a query from outside seems really complicated? Is there a better way to do each with entities that should have several components?
+		//atm I just use transform scale, since it is the same (for rectangle collision) as collision component would be. 
```

Comment:
Having multiple components in a `query` is valid as you can see in the documentation: https://www.flecs.dev/flecs/md_docs_Queries.html#autotoc_md181

This is more related to the implication of using a query. In Flecs, there are two types of queries:
- `filter`: cheap to create, efficient to iterate
- `query`: it caches the output of a filter. More expensive to create, but faster to iterate.

You can read more here: https://www.flecs.dev/flecs/md_docs_Queries.html#autotoc_md171

In this case, the issue is that the query is created, some results are cached but kinda out of order for Flecs, so it results in the crash. If you replace `ecs.query` with `ecs.filter`, this will work.